### PR TITLE
apply upcoming rule: unnecessary_lambdas

### PR DESCRIPTION
**Pull request by a14n**
[DON’T create a lambda when a tear-off will do.](https://www.dartlang.org/guides/language/effective-dart/usage#dont-create-a-lambda-when-a-tear-off-will-do)

(not sure this improves the readability of `setState` functions)

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/calculator/home.dart [hunk 1]
```diff
@@ -106,9 +106,7 @@ class _CalculatorState extends State<Calculator> {
   }
 
   void handleDelTap() {
-    setState(() {
-      popCalcExpression();
-    });
+    setState(popCalcExpression);
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/calculator/home.dart]: Please don't do this to setState. We're trying to make setState look like a block statement, where it's "obvious" that the statements inside the block are executed immediately. This change hides what's going on, it makes it look like we're taking a reference to call later or something.

File: examples/flutter_gallery/lib/demo/material/overscroll_demo.dart [hunk 2]
```diff
@@ -32,9 +32,7 @@ class OverscrollDemoState extends State<OverscrollDemo> {
          content: new Text("Refresh complete"),
          action: new SnackBarAction(
            label: 'RETRY',
-           onPressed: () {
-             _refreshIndicatorKey.currentState.show();
-           }
+           onPressed: _refreshIndicatorKey.currentState.show
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/material/overscroll_demo.dart]: This is not equivalent. What if `_refreshIndicatorKey.currentState` has changed value before the user taps the button?

File: dev/benchmarks/complex_layout/lib/main.dart [hunk 3]
```diff
@@ -611,7 +611,7 @@ class GalleryDrawer extends StatelessWidget {
           new DrawerItem(
             icon: new Icon(Icons.hourglass_empty),
             selected: timeDilation != 1.0,
-            onPressed: () { ComplexLayoutApp.of(context).toggleAnimationSpeed(); },
+            onPressed: ComplexLayoutApp.of(context).toggleAnimationSpeed,
```
Comment (Hixie) [inline dev/benchmarks/complex_layout/lib/main.dart]: This is not equivalent. The original code does the lookup at the time of the tap. The new code does the lookup at build time. The lookup is expensive. The lookup can also change result based on changes outside this function.

File: packages/flutter/test/foundation/change_notifier_test.dart [hunk 4]
```diff
@@ -187,8 +187,8 @@ void main() {
     source.dispose();
     expect(() { source.addListener(null); }, throwsFlutterError);
     expect(() { source.removeListener(null); }, throwsFlutterError);
-    expect(() { source.dispose(); }, throwsFlutterError);
-    expect(() { source.notify(); }, throwsFlutterError);
+    expect(source.dispose, throwsFlutterError);
```
Comment (Hixie) [inline packages/flutter/test/foundation/change_notifier_test.dart]: this change makes the code less readable

**Comment by Hixie** [comment]
Some of these changes are valid, but many are not.

**Comment by Hixie** [comment]
The tests caught some of them, so we've got that going for us, which is nice. :-)

**Comment by a14n** [comment]
@Hixie you're completly right. The lint has (for now) a lot of false positives. I'm going to close and I'll file issue on the linter tracker.

**Pull request closed without merging**